### PR TITLE
MPFR_jll v4.1.0 is used in Julia v1.5.1+

### PR DESCRIPTION
**Pull request by staticfloat**
Totally guessing here on the syntax for compat bounds

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Hmmm. I'm not sure this is quite what you want.

Could you elaborate a little bit on which versions of MPFR_jll should be compatible with which versions of Julia?

For example, the changes currently made in this PR result in the following situation:
| Versions of MPFR_jll | `[compat]` entry | Compatible versions of Julia | 
| -------------------- | ------------------ | ---------------------------- |
| 4.0\.* | `julia = "1-1.5.0"` | 1.0.\*, 1.1.\*, 1.2.\*, 1.3.\*, 1.4.\*, 1.5.0 |
| 4.1.\*, 4.2.\*, 4.3.\*, 4.4.\*, ..., 4.x.\* | `julia = "1.5.1"` | 1.5.1 |

---

This is how to tell which versions of Julia are included in a particular `[compat]` entry for Julia.

```julia
julia> v"1.4.0" in Pkg.Types.VersionRange("1-1.5.0")
true

julia> v"1.4.1" in Pkg.Types.VersionRange("1-1.5.0")
true

julia> v"1.4.2" in Pkg.Types.VersionRange("1-1.5.0")
true

julia> v"1.5.0" in Pkg.Types.VersionRange("1-1.5.0")
true

julia> v"1.5.1" in Pkg.Types.VersionRange("1-1.5.0")
false

julia> v"1.5.2" in Pkg.Types.VersionRange("1-1.5.0")
false

julia> v"1.6.0" in Pkg.Types.VersionRange("1-1.5.0")
false

julia> v"1.6.1" in Pkg.Types.VersionRange("1-1.5.0")
false

julia> v"1.6.2" in Pkg.Types.VersionRange("1-1.5.0")
false
```

```julia
julia> v"1.4.0" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.4.1" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.4.2" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.5.0" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.5.1" in Pkg.Types.VersionRange("1.5.1")
true

julia> v"1.5.2" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.6.0" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.6.1" in Pkg.Types.VersionRange("1.5.1")
false

julia> v"1.6.2" in Pkg.Types.VersionRange("1.5.1")
false
```

**Comment by DilumAluthge** [comment]
I think you might want to use this Julia `[compat]` entry for MPFR_jll 4.1:
```toml
julia = "1.5.1-1"
```

This results in the following:
``julia
julia> v"1.4.0" in Pkg.Types.VersionRange("1.5.1-1")
false

julia> v"1.4.1" in Pkg.Types.VersionRange("1.5.1-1")
false

julia> v"1.4.2" in Pkg.Types.VersionRange("1.5.1-1")
false

julia> v"1.5.0" in Pkg.Types.VersionRange("1.5.1-1")
false

julia> v"1.5.1" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.5.2" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.5.3" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.6.0" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.6.1" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.6.2" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.6.3" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.7.0" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.7.1" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.7.2" in Pkg.Types.VersionRange("1.5.1-1")
true

julia> v"1.7.3" in Pkg.Types.VersionRange("1.5.1-1")
true
```

File: M/MPFR_jll/Compat.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ["4.0"]
-julia = "1-1.6"
+julia = "1-1.5.0"
 
 ["4.1-4"]
-julia = "1.7"
+julia = "1.5.1"
```
Comment (DilumAluthge) [inline M/MPFR_jll/Compat.toml:5]:
```suggestion
julia = "1.5.1-1"
```

**Comment by staticfloat** [comment]
Yes, you got it. :)

**Pull request merged**